### DISCUSSION TITLE
Should we continue to support the Mapbox styles?

### DISCUSSION THREAD

**Discussion by maxammann** [maplibre/maplibre#157] [Q&A]
I was wondering if we already have an official standpoint on whether we want to continue to support the Mapbox styles or not. I think to some extend we want to support them. For instance we do not want Maplibre to crash if a Mapbox style is used.

## Levels of support

There are multiple levels (or greatness?) of support. For example we could:

* Only support non-breaking changes in the Mapbox specification
* Do not add breaking changes (in respect to the Mapbox specifiction) to the Maplibre specification (e.g. adding a property to the Maplibre specification which clashes with a new property in the Mapbox specification)
* Raise errors when users use Mapbox features which are not available in Maplibre

## Changelog for Maplibre spec

There is an official changelog for the Mapbox style: https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/CHANGELOG.md

It documents the changes, including breaking ones. We should have and maintain a similar document.

## On the Rust side

Together with @kimpham54 we are planning on creating a Rust parser for the style specification. To some extend these decisions are important for maplibre-rs. maplibre-rs will probably never supported the full Maplibre spec or the full Mapbox spec, because some features are just not used by enough people to care about them.

So at maplire-rs we need a strategy to deal with unsupported features and differences between maplibre and mapbox. Should we abort or warn with unsupported features? 

## Related

* Style Editors: #150 
* TypeScript styles: https://github.com/maplibre/maplibre-gl-js/discussions/1851

**Comment by HarelM** [comment]
I personally think that running behind Mapbox and trying to make sure we allow the new features they add to run (or at least not fail) in MapLibre is both dangerous (From legal point of view) and poor use of our time/money.
We should focus on bringing new functionality and features to MapLibre.
At least from the community point of view, if people would like to invest in conversions, parity, protocols support they can do it in plugins and external tools.
While I understand this might not be a very pleasing standpoint, Mapbox decided to change their license, I already received a comment in one of the PRs that we are violating this new license and in general, people who are using the Mapbox echo system should use Mapbox (and pay them, as they are doing a great job and should be compensated), IMHO.

**Reply by maxammann** [reply]
I completely agree with you, though we probably still need to document a strategy somewhere. E.g. do we allow breaking changes in the style spec? Which version of the style spec do we support?

Maybe we should reset the version counter to 1.0 and call it MapLibre style spec.

I just feel like we need to distance ourselves more from Mapbox :).

**Reply by maxammann** [reply]
it seems like there are multiple versions in play here. Once the style spec version which is currently `8`. And then the version of the tooling around the style spec which is currently 17.0:

https://github.com/maplibre/maplibre-gl-js/blob/main/src/style-spec/CHANGELOG.md

**Reply by wipfli** [reply]
I call it the MapLibre style spec already

**Comment by wipfli** [comment]
I think the MapLibre style specification should be about MapLibre.

The versioning in the NPM package is a bit strange but what really matters is the table which docuements what style spec feature is supported in which version of the SDKs. Feel free to add a section for maplibre-rs there by the way.

I think the MapLibre style spec should not accept breaking changes, because it is so convenient to have working styles for our users which do exactly what they wanted.

Differences between the MapLibre and the Mapbox style specification should be tracked outside of MapLibre I think, maybe in a plugin if someone is motivated.

Documemting where the MapLibre organization stands could make sense. This could be a message by the Governing Board or something the voting members approve...
